internal/otgutils/validation.go

```diff
@@ -0,0 +1,136 @@
+package otgutils
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/openconfig/ondatra"
+)
+
+// ExpectedBgpMetrics struct used for validating the fetched OTG BGP stats.
+type ExpectedBgpMetrics struct {
+	Advertised int32
+	Received   int32
+}
+
+// ExpectedIsisMetrics struct used for validating the fetched OTG ISIS stats.
+type ExpectedIsisMetrics struct {
+	L1SessionsUp   int32
+	L2SessionsUp   int32
+	L1DatabaseSize int32
+	L2DatabaseSize int32
+}
+
+// ExpectedPortMetrics struct used for validating the fetched OTG Port stats.
+type ExpectedPortMetrics struct {
+	FramesRx int32
+}
+
+// ExpectedFlowMetrics struct used for validating the fetched OTG Flow stats.
+type ExpectedFlowMetrics struct {
+	FramesRx     int64
+	FramesRxRate float32
+}
+
+// ExpectedState is used for creating expected otg metrics.
+type ExpectedState struct {
+	Port map[string]ExpectedPortMetrics
+	Flow map[string]ExpectedFlowMetrics
+	BGP4 map[string]ExpectedBgpMetrics
+	BGP6 map[string]ExpectedBgpMetrics
+	ISIS map[string]ExpectedIsisMetrics
+}
+
+// WaitForOpts is used at tests level whenever WaitFor func is called\. There are 3 parameters which could be set.
+type WaitForOpts struct {
+	Condition string
+	Interval  time.Duration
+	Timeout   time.Duration
+}
+
+// timer prints time elapsed in ms since a given start time
+func timer(t *testing.T, start time.Time, name string) {
+	elapsed := time.Since(start)
+	t.Log(name, "took", elapsed.Milliseconds(), "ms")
+}
+
+// WaitFor returns nil once the given function param returns true. It will wait and retry for the entire timeout duration.
+func WaitFor(t *testing.T, fn func() bool, opts *WaitForOpts) error {
```

Comment:
Still not convinced you should be creating WaitFor utilities when these are built into the telemetry library

Reply:
Replace with built-in utilities